internal/linker/link_node_modules.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @fileoverview Creates a node_modules directory in the current working directory
+ * and symlinks in the node modules needed to run a program.
+ * This replaces the need for custom module resolution logic inside the process.
+ */
+const fs = require('fs');
+const path = require('path');
+
+function debug(...m) {
+  if (!!process.env['VERBOSE_LOGS']) console.error('[link_node_modules.js]', ...m);
+}
+
+function ls_r(d) {
+  for (const p of fs.readdirSync(d)) {
+    const f = path.join(d, p);
+    debug('  ', f);
+    if (fs.statSync(f).isDirectory()) {
```

Comment:
could use the new readdir Dirent api here.
also potential for infinite recursion with symlinks that link to a parent. change to lstat or use dirent.isSymbolicLink

Reply:
ditto, let's just drop this code